eve/validation.py

```diff
@@ -12,5 +12,144 @@
+
+        .. versionadded:: 0.1.0
+        """
+        self.document_id = document_id
+        self.persisted_document = persisted_document
+        return super(Validator, self).validate(document)
+
+    def _normalize_default(self, mapping, schema, field):
+        """ {'nullable': True} """
+        if not self.persisted_document or \
+           field not in self.persisted_document:
+            super(Validator, self)._normalize_default(mapping, schema, field)
+
+    def _normalize_default_setter(self, mapping, schema, field):
+        """ {'oneof': [
+                {'type': 'callable'},
+                {'type': 'string'}
+                ]} """
+        if not self.persisted_document or \
+           field not in self.persisted_document:
+            super(Validator, self)._normalize_default_setter(mapping, schema,
+                                                             field)
+
+    def _validate_dependencies(self, dependencies, field, value):
+        """ {'type': ['dict', 'hashable', 'hashables']} """
+        persisted = self._filter_persisted_fields_not_in_document(dependencies)
+        if persisted:
+            dcopy = copy.copy(self.document)
+            for field in persisted:
+                dcopy[field] = self.persisted_document[field]
+            validator = self._get_child_validator()
+            validator.validate(dcopy, update=self.update)
+            self._error(validator._errors)
+        else:
+            super(Validator, self)._validate_dependencies(dependencies, field,
+                                                          value)
+
+    def _filter_persisted_fields_not_in_document(self, fields):
+        def persisted_but_not_in_document(field):
+            return field not in self.document and \
+                self.persisted_document and \
+                field in self.persisted_document
+        return list(filter(persisted_but_not_in_document, fields))
+
+    def _validate_readonly(self, read_only, field, value):
+        """ {'type': 'boolean'} """
+        persisted_value = self.persisted_document.get(field) \
+            if self.persisted_document else None
+        if value != persisted_value:
+            super(Validator, self)._validate_readonly(read_only, field, value)
+
+    @property
+    def resource(self):
+        return self._config.get('resource', None)
+
+    @resource.setter
+    def resource(self, value):
+        self._config['resource'] = value
+
+    @property
+    def document_id(self):
+        return self._config.get('document_id', None)
+
+    @document_id.setter
+    def document_id(self, value):
+        self._config['document_id'] = value
+
+    @property
+    def persisted_document(self):
+        return self._config.get('persisted_document', None)
+
+    @persisted_document.setter
+    def persisted_document(self, value):
+        self._config['persisted_document'] = value
+
+
+class DefaultErrorHandler(cerberus.errors.BasicErrorHandler):
```

Comment:
I agree that we could do this check only one time at application startup. But I wonder if creating a dynamic type based on a config item can get confusing once someone will subclass this again.

How about renaming `DefaultErrorHandler` to something like `SingleErrorsAsStringErrorHandler` (or a better name if you can come with one) and then amend ```Validator.__init__``` as follows?
  ```python
  if not config.VALIDATION_ERROR_AS_LIST:
      kwargs['error_handler'] = SingleErrorsAsStringErrorHandler
  ```

But again I cannot see a "real" issue with your (definitely more sophisticated) suggestion.